agent/background.go

```diff
@@ -129,3 +131,22 @@ func (a *Agent) withCallbackTimeout(ctx context.Context, tc *taskContext) (conte
 	}
 	return context.WithTimeout(ctx, timeout)
 }
+
+func (a *Agent) startSpotTerminationWatcher(ctx context.Context, tc *taskContext, cancel context.CancelFunc) {
+	defer recovery.LogStackTraceAndContinue("spot termination watcher")
+	defer cancel()
+	ticker := time.NewTicker(10 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			grip.Info("Spot termination watcher canceled")
+			return
+		case <-ticker.C:
+			if util.SpotHostWillTerminateSoon() {
+				grip.Info("Spot instance terminating, so agent is exiting")
+				os.Exit(1)
```

Comment:
Is it safe to exit this hard? I wonder if this will leave unreleased resources (for example the log file might be left open). Should this cancel its context?

Reply:
I think it's okay to exit this hard, since the host will be terminated within 2 minutes at this point. Essentially I want the agent to stop immediately, mainly so that spurious failures in the task as a result of the host shutting down aren't propagated to the app server. The state of the host no longer matters, since it's about to shut down.